upgrade/helpers/tasks.py

```diff
@@ -765,35 +767,48 @@ def upgrade_using_foreman_maintain():
-            run('foreman-maintain upgrade check --target-version {}'
-                ' -y'.format(os.environ.get('TO_VERSION')))
-
-    # whitelist disk-performance check
-    if os.environ.get('FROM_VERSION') == os.environ.get('TO_VERSION'):
-        # z stream upgrade
-        run('foreman-maintain upgrade run '
-            '--whitelist="disk-performance" '
-            '--target-version {} '
-            '-y'.format(os.environ.get('TO_VERSION') + ".z"))
-    else:
-        # use beta until 6.7 is GA
-        if os.environ.get('TO_VERSION') == '6.7':
-            with shell_env(FOREMAN_MAINTAIN_USE_BETA='1'):
-                if bz_bug_is_open(1781128):
-                    run('foreman-maintain upgrade run '
-                        '--whitelist="disk-performance, foreman-tasks-not-paused" '
-                        '--target-version {} -y'.format(os.environ.get('TO_VERSION')))
-                else:
-                    run('foreman-maintain upgrade run '
-                        '--whitelist="disk-performance" '
-                        '--target-version {} -y'.format(os.environ.get('TO_VERSION')))
+            run(f'foreman-maintain upgrade check --target-version'
+                f' {os.environ.get("TO_VERSION")} -y')
+
+    def satellite_upgrade():
+        """ This inner function is used to perform Y & Z satellite stream upgrade"""
+        # whitelist disk-performance check
+        if os.environ.get('FROM_VERSION') == os.environ.get('TO_VERSION'):
+            # z stream satellite upgrade
+            run(f'foreman-maintain upgrade run '
+                f'--whitelist="disk-performance" '
+                f'--target-version {os.environ.get("TO_VERSION") + ".z"} -y')
         else:
-            run('foreman-maintain upgrade run '
-                '--whitelist="disk-performance" '
-                '--target-version {} -y'.format(os.environ.get('TO_VERSION')))
+            # use beta until 6.8 is GA
+            if os.environ.get('TO_VERSION') == '6.8':
+                with shell_env(FOREMAN_MAINTAIN_USE_BETA='1'):
+                    run(f'foreman-maintain upgrade run --whitelist="disk-performance" '
+                        f'--target-version {os.environ.get("TO_VERSION")} -y')
+            else:
+                run(f'foreman-maintain upgrade run --whitelist="disk-performance" '
+                    f'--target-version {os.environ.get("TO_VERSION")} -y')
+
+    def capsule_upgrade():
+        """ This inner function is used to perform Y & Z stream Capsule upgrade"""
+        if os.environ.get('FROM_VERSION') == os.environ.get('TO_VERSION'):
+            # z capsule stream upgrade
+            run(f'foreman-maintain upgrade run --target-version '
+                f'{os.environ.get("TO_VERSION") + ".z"} -y')
+        else:
+            if os.environ.get('TO_VERSION') == '6.8':
+                with shell_env(FOREMAN_MAINTAIN_USE_BETA='1'):
+                    run(f'foreman-maintain upgrade run --whitelist="repositories-validate, '
```

Comment:
Why are we skipping repositories-validate, repositories-setup? we should follow same process as satellite upgrade right?

Reply:
If we do not skip then "foreman-maintan run" failed with below message
```
# foreman-maintain upgrade check --target-version 6.7 -y
"The following steps ended up in failing state:
[repositories-validate]
Resolve the failed steps and rerun the command. In case the failures are false positives,
```